codegen/module.go

```diff
@@ -119,12 +119,16 @@ func (system *ModuleSystem) RegisterClass(
 	return nil
 }
 
+// ModuleClassOption allows customization with the module class.
+type ModuleClassOption func(*ModuleClass)
+
 // RegisterClassType registers a type generator for a specific module class
 // For example, the "http"" type generator for the "Endpoint"" class
 func (system *ModuleSystem) RegisterClassType(
 	className string,
 	classType string,
 	generator BuildGenerator,
+	opts ...ModuleClassOption,
```

Comment:
I generally dislike this approach. We see this sort of code littered around our codebases where he have "preHandleHooks" and stuff. The main issues are that it doesn't document itself and it relies on mutation. The class types should be clearly configurable in one way.

My original suggestion was to make the nested directory a static property of the ClassType definition. So for 
the Class "Endpoint" you can register a ClassType "http", and for that type you can optionally choose the subdirectory for that type.